### PR TITLE
Initial work on symmetric eigensolvers

### PR DESCRIPTION
**Pull request by kshyatt**
Add in some of the new methods for symmetric (generalized or simple) eigensolving.

### CODE REVIEW

**Comment by maleadt** [comment]
Feel free to call in bors to `try` or try & merge (`r+`) the PR when you think it is ready.
bors try

File: test/solver.jl [hunk 1]
```diff
@@ -190,6 +190,22 @@ k = 1
...
+        if( elty <: Complex )
+            d_W, d_V   = CUSOLVER.heevd!('V','U', d_A)
+        else
+            d_W, d_V   = CUSOLVER.syevd!('V','U', d_A)
+        end
+        h_W            = collect(d_W)
+        h_V            = collect(d_V)
+        Eig            = eigfact(A)
+        @test Eig[:vals] ≈ h_W
+        @test Eig[:vecs] ≈ h_V
```
Comment (andreasnoack) [inline test/solver.jl]: You might need to account for the phase here. Something like `abs.(Eig[:vecs]'*h_V) ≈ I` might be sufficient.

File: src/solver/dense.jl [hunk 2]
```diff
@@ -370,3 +378,91 @@ for (bname, fname, elty, relty) in ((:cusolverDnSgesvd_bufferSize, :cusolverDnSg
         end
     end
 end
+
+for (jname, bname, fname, elty, relty) in ((:syevd!, :cusolverDnSsyevd_bufferSize, :cusolverDnSsyevd, :Float32, :Float32),
+                                           (:syevd!, :cusolverDnDsyevd_bufferSize, :cusolverDnDsyevd, :Float64, :Float64),
+                                           (:heevd!, :cusolverDnCheevd_bufferSize, :cusolverDnCheevd, :Complex64, :Float32),
```
Comment (andreasnoack) [inline src/solver/dense.jl]: `Complex64` -> `ComplexF32` and similar for `Complex128`.

**Comment by andreasnoack** [comment]
bors try

File: src/solver/dense.jl [hunk 3]
```diff
@@ -370,3 +378,91 @@ for (bname, fname, elty, relty) in ((:cusolverDnSgesvd_bufferSize, :cusolverDnSg
...
+for (jname, bname, fname, elty, relty) in ((:syevd!, :cusolverDnSsyevd_bufferSize, :cusolverDnSsyevd, :Float32, :Float32),
+                                           (:syevd!, :cusolverDnDsyevd_bufferSize, :cusolverDnDsyevd, :Float64, :Float64),
+                                           (:heevd!, :cusolverDnCheevd_bufferSize, :cusolverDnCheevd, :ComplexF32, :Float32),
+                                           (:heevd!, :cusolverDnZheevd_bufferSize, :cusolverDnZheevd, :ComplexF64, :Float64))
+    @eval begin
+        function $jname(jobz::Char,
+                        uplo::Char,
+                        A::CuMatrix{$elty})
+            cuuplo  = cublasfill(uplo)
+            cujobz  = cusolverjob(uplo)
```
Comment (andreasnoack) [inline src/solver/dense.jl]: `uplo` -> `jobz`

**Comment by andreasnoack** [comment]
bors try

**Comment by andreasnoack** [comment]
Maybe it’s easier if you make tests pass locally.

tor. 13. dec. 2018 kl. 17.21 skrev bors[bot] <notifications@github.com>:

**Comment by kshyatt** [comment]
They are! That's what I don't get, I'm running the tests on my local cluster and they are passing.

**Comment by kshyatt** [comment]
bors try

**Comment by kshyatt** [comment]
good to squash and merge?

**Pull request merged**